src/controller/worryController.ts

```diff
@@ -0,0 +1,25 @@
+import { Request, Response } from "express";
+import { fail, success } from "../constants/response";
+import { rm, sc } from "../constants";
+import { ClientException } from "../common/error/exceptions/customExceptions";
+import statusCode from "../constants/statusCode";
+import { worryService } from "../service";
+
+const getWorryListByCategory = async (req: Request, res: Response) => {
+    try {
+        const { categoryId } = req.params;
+
+        if (!categoryId) {
+            throw new ClientException("필요한 Param 값이 없습니다.");
+        }
+
+        const data = await worryService.getWorryListByCategoryId(+categoryId);
+
+        return res.status(sc.OK).send(success(statusCode.OK, rm.READ_WORRYLIST_SUCCESS, data));
+
+    } catch (error) {
+        return res.status(sc.BAD_REQUEST).send(fail(sc.BAD_REQUEST, rm.READ_WORRYLIST_FAIL));
```

Comment:
p1. 작성해주신 코드는 controller 하위에서 발생한 모든 예외에 대해 같은 응답을 보내게 되는 문제가 있어요.

```suggestion
const getWorryListByCategory = async (req: Request, res: Response, next: NextFunction) => {
   // ...
    try {
        // ...
    } catch (error) {
        next(error);
       }
   }
```
이렇게 하면, 미들웨어에 등록돼있는 에러 핸들러(/common/error/handler.ts) 에서 응답을 보내주어요.
우리가 예상한 예외가 아닌 경우에도 처리를 해주고, 우리가 던진 예외마다 정의한 status code로 응답해줄 수 있어요.

만약 우리가
```typescript
throw new ClientExcpetion("걱정 글 삭제 권한이 없습니다.", 403);
```
이렇게 예외를 던진다면, 우리의 에러 핸들러가
```typescript
res.status(status).send(fail(403, "걱정 글 삭제 권한이 없습니다."));
```
이렇게 응답해줘용

Reply:
와 상세한 답변 감사해요 .. 에러핸들러 짱짱이네요 ..